src/snowflake/cli/_plugins/snowpark/package_utils.py

```diff
@@ -222,23 +215,30 @@ def pip_wheel(
     if not dependencies:
         command.append("--no-deps")
 
-    try:
+    log.info(
+        "Running pip wheel with command: %s",
+        " ".join([str(com) for com in command]),
+    )
+    result = subprocess.run(
+        ["python", *command],
+        capture_output=True,
+        text=True,
+        encoding=locale.getpreferredencoding(),
+    )
+    if result.returncode != 0:
         log.info(
-            "Running pip wheel with command: %s",
-            " ".join([str(com) for com in command]),
-        )
-        process = subprocess.run(
-            ["python", *command],
-            capture_output=True,
-            text=True,
-            encoding=locale.getpreferredencoding(),
+            "Pip wheel finished with error code %d. Details: %s",
+            result.returncode,
+            result.stdout + result.stderr,
         )
-    except subprocess.CalledProcessError as e:
-        log.error("Encountered error %s", e.stderr)
-        raise ClickException(f"Encountered error while running pip wheel.")
+        if raise_on_error:
+            raise ClickException(
+                f"pip wheel finished with error code {result.returncode}. Please re-run with --verbose or --debug for more details."
+            )
+    else:
+        log.info("Pip wheel command executed successfully")
```

Comment:
Do we want spam this message?

Reply:
I kept that as I liked the log consistency - "running pip" -> "pip error" or "running pip" -> "pip success", no strong opinion though